src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -149,6 +179,18 @@ ClassLoaderData* SystemDictionary::register_loader(Handle class_loader, bool cre
   }
 }
 
+void SystemDictionary::set_system_loader(ClassLoaderData *cld) {
+  if (_java_system_loader.is_empty()) {
+    _java_system_loader = cld->class_loader_handle();
+  }
```

Comment:
Wouldn't it be an error to call this when the loader has already been set?

Reply:
Hi David,

Are you thinking about adding an `else` case to the function and call the `vm_exit_during_initialization()` ?

Reply:
I was thinking:
```
void SystemDictionary::set_system_loader(ClassLoaderData *cld) {
  assert(_java_system_loader.is_empty(), "already set!");
  _java_system_loader = cld->class_loader_handle();
}
```

Reply:
Wait, I just realized, how does this not get into trouble with GC?  If we copy this OopHandle, then GC will follow it from two locations.  This doesn't seem right.

Reply:
Hi Coleen,

Let me know what needs to be done to get this right. I can file another bug to fix it together with David's suggestion above.

Reply:
Sorry Calvin, this is fine. We can copy OopHandle because they're walked by their OopStorage location not where they're copied to anymore. We do this in lots of places.

Reply:
Thanks Coleen.